File: android_code/ETCB/factory/src/main/java/club/crabglory/www/factory/Factory.java
package club.crabglory.www.factory;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

public class Factory {

    private final ExecutorService executor;
    private static Factory factory;

    public Factory() {
        executor = Executors.newFixedThreadPool(4);
    }

    public static void runOnAsync(Runnable runnable){
        Factory.getInstance().executor.execute(runnable);
    }

    static {
        factory = new Factory();
    }
    static Factory getInstance(){
        return factory;
    }
}

File: android_code/ETCB/factory/src/main/java/club/crabglory/www/factory/Factory.kt
package club.crabglory.www.factory

import club.crabglory.www.data.DataKit
import club.crabglory.www.data.model.db.utils.DBFlowExclusionStrategy
import com.google.gson.Gson
import com.google.gson.GsonBuilder
import java.util.concurrent.Executors

class Factory{
    companion object {
        private var executor = Executors.newFixedThreadPool(4)!!
        private val gson: Gson

        init {
            executor = Executors.newFixedThreadPool(4)
            gson = GsonBuilder() // 设置时间格式
                    .setDateFormat("yyyy-MM-dd'T'HH:mm:ss") // 设置一个过滤器，数据库级别的Model不进行Json转换
                    .setExclusionStrategies(DBFlowExclusionStrategy())
                    .create()
        }

        fun runOnAsync(runnable: Runnable) {
            executor.submit(runnable)
        }

        fun getGson(): Gson {
            return gson
        }
    }
}